frontend/src/concepts/projects/apiHooks/useNamespaceSecret.ts

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import { getSecret } from '~/api';
+import { EnvVariable, EnvironmentVariableType, SecretCategory } from '~/pages/projects/types';
+import useFetchState, { FetchState } from '~/utilities/useFetchState';
+
+const useNamespaceSecret = (
```

Comment:
Hah, we (https://github.com/opendatahub-io/odh-dashboard/pull/1195) are doing the same thing here -- we'll need to align on one of our functions after we both merge.